components/_impl/workers/subprocess_rpc.py

```diff
@@ -138,20 +150,15 @@ def pop(self, w_fd: int) -> None:
     def maybe_timeout_read(cls, pipe: "Pipe") -> None:
         timeout = pipe.timeout
 
-        # Workers should never set a timeout, so in that case we want to exit
-        # without calling `cls.singleton()` so we don't spawn an unnecessary
-        # loop thread.
-        if timeout is None:
-            yield
-            return
-
+        # Spawn a loop thread to periodically check the liveness of subprocess
         w_fd = pipe.write_fd
-        assert w_fd is not None, "Cannot timeout without write file descriptor."
-        singleton = cls.singleton()
+        # assert w_fd is not None, "Cannot timeout without write file descriptor."
```

Comment:
TODO: removing this assertion is bad, but still trying to understand why `w_fd` could be `None` here. @robieta any ideas?

Reply:
It will be None if this pipe can only be used to read. The child only gets the read_fd for the input pipe, and only gets the write_fd for the output and data pipes.

Reply:
Thanks! Now the timeout and zombie status check is only for the parent process and `output_pipe`.